Userland/Libraries/LibJS/Runtime/VM.cpp

```diff
@@ -262,52 +263,60 @@ void VM::assign(const NonnullRefPtr<BindingPattern>& target, Value value, Global
             if (exception())
                 return;
 
-            if (pattern_property.name) {
-                set_variable(pattern_property.name->string(), value, global_object, first_assignment, specific_scope);
-                if (pattern_property.is_rest)
-                    break;
-                continue;
-            }
+            auto& alias = pattern_property.alias;
+            if (!alias.has<Empty>()) {
```

Comment:
This chain of ifs will check the index multiple times, and is a bit too verbose
Could use `visit` to match the type instead
```c++
alias.visit(
    [&](Empty) {},
    [&](NonnullRefPtr<Identifier> const& ident) { set_variable(ident->string(), ...); },
    [&] (BindingPattern const& pattern) { assign(pattern, ...); });
```